.github/workflows/unit-tests.yml

```diff
@@ -48,3 +48,16 @@ jobs:
             pytest-report-${{ matrix.python }}.xml
             htmlcov-${{ matrix.python }}/
           retention-days: 7
+
+      - name: Check coverage
+        run: |
+          coverage=$(uv run --python ${{ matrix.python }} --group unit coverage report -m | tail -1 | tail -c 4 | head -c 2)
+          echo "Total coverage: ${coverage}%"
+
+          # Check if coverage is below the threshold
+          if (( ${coverage} < 40 )); then
+              echo "Coverage failed at ${coverage}%, expected at least 40%"
+              exit 1
+          else
+              echo "Coverage check passed with ${coverage}%"
+          fi
```

Comment:
Why do we do this?

Reply:
It's a tradeoff in exchange for more test coverage. We can remove the check, and maybe we could revisit this in the future.